ruby/lib/gherkin/query.rb

```diff
@@ -20,34 +27,42 @@ def update_feature(feature)
       store_nodes_location(feature.tags)
 
       feature.children.each do |child|
-        update_rule(child.rule) if child.rule
-        update_background(child.background) if child.background
-        update_scenario(child.scenario) if child.scenario
+        update_rule(feature, child.rule) if child.rule
+        update_background(feature, child.background) if child.background
+        update_scenario(feature, nil, child.scenario) if child.scenario
```

Comment:
5second check. Should this be `child&rule` to cover where the child does / doesn't respond to rule?

Reply:
Right, so this should be added then right?

Reply:
Hmm, I'm not following. Are you saying that we might want to change line 30 by adding a safe navigation operator (&.) to the postfix guard clause, like so?

`update_rule(feature, child.rule) if child&.rule`

Reply:
No the line I've commented on here - 32. you have a `nil` rule. this feels wrong?

Reply:
@luke-hill Ahh, I gotcha now. Let me dig in and see if I can figure out why I thought `nil` made sense back when I wrote this.

Reply:
@luke-hill I really can't figure out why I put `nil` there. All the tests still pass when replacing it with `child.rule`. No need for safe nav operator I don't think, since child always responds to `#rule` (it just might be nil, which is handled in the called method). I'll push a commit.

Reply:
@luke-hill Okay, pushed. Could probably do with a squash when merging.